h2/src/main/org/h2/mode/FunctionsPostgreSQL.java

```diff
@@ -221,6 +231,29 @@ protected Value getValueWithArgs(Session session, Expression[] args) {
             // Not implemented
             result = v1.convertTo(Value.VARCHAR);
             break;
+        case ARRAY_TO_STRING:
+            StringJoiner joiner = new StringJoiner(v1.getString());
+            if (v0.getValueType() != Value.ARRAY) {
```

Comment:
You need to return `ValueNull.INSTANCE` if `v0` or `v1` are equal to `ValueNull.INSTANCE` before this condition, because this function now processes nulls by itself.

Reply:
Please, also add test cases for `NULL` values in the first and second arguments to ensure correctness of the new condition.

Reply:
I adjusted test cases:

- array_to_string(null, ?, ?) -> null
- array_to_string(?, null, ?) -> null
- array_to_string(<not-array>, ?, ?) -> invalid

Although PostgreSQL will regard `array_to_string(null, <string>)` as invalid.

Reply:
You need to add an explicit cast to array data type when you test `NULL` array in the PostgreSQL, it rejects untyped `NULL` only.

Reply:
You mean `array_to_string(null::array, ',')` should be considered?

Reply:
`int array` or some other valid array type. In our tests it isn't needed, I meant only how you can test it in the real PostgreSQL correctly.

Reply:
```
select array_to_string(array[1, null, 3], 0, 2);
select array_to_string(null, 0, 2);
select array_to_string(array[1, null, 3], null, 2);
select array_to_string(0, ',');
```

These are rejected as `function does not exist` by PostgreSQL 12.  Seems PG matches function by name and type.